crates/cli/src/main.rs

```diff
@@ -7,15 +7,25 @@ mod source_code_section;
 use crate::options::Options;
 use anyhow::{bail, Context, Result};
 use std::env;
+use std::fs::File;
 use std::io::{Read, Write};
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::process::Stdio;
 use std::{fs, process::Command};
 use structopt::StructOpt;
 
+const QUICKJS_PROVIDER_MODULE: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/../../target/wasm32-wasi/release/javy_quickjs_provider.wasm"
+));
+
 fn main() -> Result<()> {
     let opts = Options::from_args();
 
+    if let Some(path) = &opts.provider {
```

Comment:
This is a bit iffy because it will only emit the binary if the rest of the invocation is valid. SO just `javy --emit-provider` fails. But I am not sure how to teach StructOpt to behave differently...

Reply:
Arguably some way of expressing a subcommand (for example, `javy emit-provider`) would be a way to do this so the flags are tied to a subcommand instead of Javy in general. But that's a more disruptive change.

Reply:
I'd prefer if we add a subcommand for this, since it is unrelated to the main command which is compilation.